service/round.go

```diff
@@ -65,15 +65,14 @@ type round struct {
 
 	stateCache *roundState
 
-	sig       *signal.Signal
 	submitMtx sync.Mutex
 }
 
 func (r *round) Epoch() uint32 {
 	return r.execution.Epoch
 }
 
-func newRound(sig *signal.Signal, datadir string, epoch uint32) *round {
+func newRound(ctx context.Context, serverGroup *errgroup.Group, datadir string, epoch uint32) *round {
```

Comment:
same here, I think a round should have its own `errgroup` instead of having it passed in from the parent.